src/pint/pint_matrix.py

```diff
@@ -72,16 +79,65 @@ def label_units(self):
             units.append(self.get_axis_labels(dim))
         return units
 
-    def get_label_size(self, label):
-        """Get the size of the a label in each axis.
+    def get_label_names(self, axis=None):
+        """return only the names of the labels
+        along the specified axis if requested
+
+        Parameters
+        ----------
+        axis: int (optional)
+            axis for the label lookup
+
+        Returns
+        -------
+        labels : list of label names for each dimension
```

Comment:
For each dimension or just  the chosen axis?

Reply:
updated